21.py

```diff
@@ -0,0 +1,52 @@
+# https://projecteuler.net/problem=21
+
+def is_proper_divisor(divisor, n):
+    if divisor >= n:
+        return False
+    if n % divisor == 0:
+        return True
+    return False
+
+
+def sum_proper_divisors(n):
+    sum = 0
+
+    for x in range(1, n + 1):
```

Comment:
divisors are always in pairs so there's no need to check the whole 1...n list

thinking of 36 as an example, the divisors are:
(i'm not sure if proper divisor includes 1 and the number but ignore them if it needs to be ignored)
1, 2, 3, 4, 
6, (6), 
9, 12, 18, 36

with this is mind, you really only need to check 1..sqrt(n)

Reply:
🤠